include/alpaka/acc/AccOacc.hpp

```diff
@@ -0,0 +1,511 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#if _OPENACC < 201306
+    #error If ALPAKA_ACC_ANY_BT_OACC_ENABLED is set, the compiler has to support OpenACC xx or higher!
```

Comment:
What is `OpenACC xx`?

Reply:
I never checked which version of OpenACC would be required as a minimum and it is not really possible to check, as no compiler actually works for all tests.

Also technically, no version of OpenACC is sufficient due to the `syncBlockThreads` issue, which requires a barrier implemented based in atomics, which is not allowed by the standard.

Reply:
I see, that makes sense. Could you maybe add a //TODO that this value needs to be set and why it is not set yet?

Reply:
That should be a doxygen comment then, right? i.e. `//! \todo ...`

Reply:
The value actually checked for amounts to OpenACC 2.0, btw. which may or may not be actually sufficient. At the moment this is more to be a check whether the compiler claims any OpenACC support. I usually used the 2.7 or 3.0 documentation, but checking for these versions would break building with NVHPC, which supports at most 2.6 atm (cannot find documentation on this one now, but I know that it does claim support 2.7).

I could either:
1. Put 2.0 in the error message to make it consistent and have the message tell the truth at leas about the check, or
2. put a `\todo` about which I would not be sure when it could an would be addressed.

I tend to towards 1. now: When something fails because of an actual support issue, the version check will be updated as a bugfix either way.

Reply:
No doxygen comment. This is a comment for developers that you can grep for or your IDE shows somewhere in a TODO list. I do not see a benefit of making this comment appear in a documentation.

You can then leave the error message as is.

Reply:
I do not know if clang-format can check this, but maybe this should be a point of discussion in https://github.com/alpaka-group/alpaka/pull/1194 . Alpaka contains all kinds of `// TODO` `// \TODO` `//! \TODO` `//@ \todo` ...

Reply:
I changed the message, but without comment. This does not need to be in a todo list. It will come up if and when it get relevant.